chain/network/src/peer/peer_actor.rs

```diff
@@ -719,135 +742,147 @@ impl PeerActor {
+                                                conn.send_message(Arc::new(
+                                                    PeerMessage::RequestUpdateNonce(PartialEdgeInfo::new(
+                                                        &network_state.config.node_id(),
+                                                        &conn.peer_info.id,
+                                                        Edge::create_fresh_nonce(&clock),
+                                                        &network_state.config.node_key,
+                                                    )
+                                                )));
+
+                                            }
+                                        }
+                                    });
+                                }
+                                // Sync the RoutingTable.
+                                act.sync_routing_table();
+                                // Sync snapshot hosts
+                                act.sync_snapshot_hosts();
                             }
-                            // Sync the RoutingTable.
-                            act.sync_routing_table();
-                            // Sync snapshot hosts
-                            act.sync_snapshot_hosts();
+
+                            #[cfg(test)]
+                            act.network_state.config.event_sink.send(Event::HandshakeCompleted(HandshakeCompletedEvent{
+                                stream_id: act.stream_id,
+                                edge: edge_clone,
+                                tier: conn.tier,
+                            }));
+                        },
+                        Err(err) => {
+                            tracing::info!(target: "network", "{:?}: Connection with {:?} rejected by PeerManager: {:?}", act.my_node_id(),conn.peer_info.id,err);
+                            act.stop(ClosingReason::RejectedByPeerManager(err));
                         }
+                    }
 
-                        #[cfg(test)]
-                        act.network_state.config.event_sink.send(Event::HandshakeCompleted(HandshakeCompletedEvent{
-                            stream_id: act.stream_id,
-                            edge: edge_clone,
-                            tier: conn.tier,
-                        }));
-                    },
-                    Err(err) => {
-                        tracing::info!(target: "network", "{:?}: Connection with {:?} rejected by PeerManager: {:?}", act.my_node_id(),conn.peer_info.id,err);
-                        act.stop(ctx,ClosingReason::RejectedByPeerManager(err));
+                    // Execute all actions that were delayed while registration was in progress.
+                    // Technically this should always be a Some, but let's be defensive.
+                    if let Some(buffered_actions) = act.registration_buffered_actions.take() {
```

Comment:
Ah I understand now why Option is there.. Can you either document it or encode it better? e.g.
``` enum RegistrationStatus { Done, Pending, InProgress(actions } ```